pkg/ingester/metrics.go

```diff
@@ -496,6 +512,29 @@ func newTSDBMetrics(r prometheus.Registerer) *tsdbMetrics {
 			"Total number of TSDB checkpoint creations attempted.",
 			nil, nil),
 
+		tsdbExemplarsTotal: prometheus.NewDesc(
```

Comment:
[question, not statement] Do we need all these metrics to be per user? We try to keep cardinality of exposed metrics low, so I'm wondering if we actually need it on a per user basis.

Reply:
Because exemplars are so new, it seemed reasonable to expose as much information as possible for tuning and troubleshooting.   We can discuss making some of the metrics global instead of per-user.   I think `prometheus_tsdb_exemplar_last_exemplars_timestamp_seconds` should remain per-user because it will help understand the relation between number of exemplars in storage and length of history for the user's workload.

Reply:
I see your point and I agree the more granularity the better from an observability perspective. We also have clusters with a large number of tenants per ingester (several thousands tenants / ingester) so this can easily end up in additional significative metrics exported by Cortex.

That's why I was asking if all of them are required. We typically start small and split by user if it turns out the metric is not useful unless splitted by user.

I think the following could be global to have a sense of the rate:
- `cortex_ingester_tsdb_exemplar_exemplars_appended_total`
- `cortex_ingester_tsdb_exemplar_exemplars_in_storage`
- `cortex_ingester_tsdb_exemplar_series_with_exemplars_in_storage`
- `cortex_ingester_tsdb_exemplar_out_of_order_exemplars_total`

That being said, up to you. Not a blocker!

Reply:
I propose the following split, which should provide the most useful information per-tenant:

```
Per-user:
cortex_ingester_tsdb_exemplar_last_exemplars_timestamp_seconds
cortex_ingester_tsdb_exemplar_series_with_exemplars_in_storage
distributor_exemplars_in_total

Global:
cortex_ingester_tsdb_exemplar_exemplars_appended_total
cortex_ingester_tsdb_exemplar_exemplars_in_storage
cortex_ingester_tsdb_exemplar_out_of_order_exemplars_total
```

Reply:
This gauge is the timestamp of the oldest exemplar currently in the circular buffer. By computing `time() - cortex_ingester_tsdb_exemplar_last_exemplars_timestamp_seconds` we can determine how much effective exemplar history is being stored.  Because tenant workloads and exemplar usage are expected to vary, I believe this metric will be important to determining appropriate values for `max_exemplars`.